bump.lua

```diff
@@ -194,6 +194,11 @@ local function rect_detectCollision(x1,y1,w1,h1, x2,y2,w2,h2, goalX, goalY)
   }
 end
 
+-- in the shadow of at least one side of rect
```

Comment:
If you stretch the edges of a rectangle to the infinite, that defines a "cross-like" region. 

So this:
```
                                                                          
   ******                                                                 
   *    *                                                                 
   *    *                                                                 
   *    *                                                                 
   ******                                                                 
             
```

Has the following "shadow":
```
   *    *                                                                 
   *    *                                                                 
   *    *                                                                 
 *******************                                                      
   *    *                                                                 
   *    *                                                                 
   *    *                                                                 
 *******************                                                      
   *    *                                                                 
   *    *                                                                 
   *    *                                                                 
   *    *                                                                 
```        

`shadow` means "the other rectangle is touching this rectangle's cross".